src/eventstore/EventStoreSubscriber.cs

```diff
@@ -5,125 +5,72 @@
 namespace CorshamScience.MessageDispatch.EventStore
 {
     using System;
-    using System.Collections.Concurrent;
     using System.Linq;
     using System.Threading;
-    using System.Timers;
+    using System.Threading.Tasks;
     using CorshamScience.MessageDispatch.Core;
-    using global::EventStore.ClientAPI;
-    using Timer = System.Timers.Timer;
+    using global::EventStore.Client;
+    using Microsoft.Extensions.Logging;
 
     /// <summary>
     /// Subscriber for event store.
     /// </summary>
     public class EventStoreSubscriber
     {
-        private const string HeartbeatEventType = "SubscriberHeartbeat";
-
-        private readonly Timer _liveProcessingTimer = new Timer(TimeSpan.FromMinutes(10).TotalMilliseconds);
-        private readonly string _heartbeatStreamName = $"SubscriberHeartbeat-{Guid.NewGuid()}";
-
         private readonly WriteThroughFileCheckpoint _checkpoint;
         private readonly object _subscriptionLock = new object();
 
-        private IEventStoreConnection _connection;
-        private long? _startingPosition;
-        private int _maxLiveQueueSize;
-        private object _subscription;
-        private int _catchupPageSize;
+        private EventStoreClient _eventStoreClient;
+        private ulong? _startingPosition;
+        private StreamSubscription _subscription;
         private string _streamName;
         private bool _liveOnly;
-
-        private long _lastNonLiveEventNumber = long.MinValue;
-        private long? _lastReceivedEventNumber;
+        private bool _isSubscribed;
+        private ulong? _lastProcessedEventNumber;
         private int _eventsProcessed;
-        private bool _catchingUp = true;
-
-        private TimeSpan _heartbeatTimeout;
-        private DateTime _lastHeartbeat;
-        private Timer _heartbeatTimer;
-        private bool _usingHeartbeats;
+        private ulong _liveEventThreshold;
```

Comment:
Mentioned elsewhere, but given our checkpoints are limited to longs should we just have this whole system use longs?